### PR TITLE
Point: Adds ANCHOR_XXX constants like ANCHOR_MIDDLE, ANCHOR_TOP_RIGHT, e...

### PR DESCRIPTION
**Pull request by WenhaiLin**
...tc.

### CODE REVIEW

**Comment by minggo** [comment]
New pull request for #3859.

**Comment by ricardoquesada** [comment]
thanks.

**Comment by winipcfg** [comment]
Isn't it better to use 1.0f, 0.5f, etc instead?

**Comment by minggo** [comment]
@winipcfg 
What did you mean?

**Comment by ricardoquesada** [comment]
@winipcfg 
you can still use 1.0 or 0.5 if you want.

**Comment by minggo** [comment]
@winipcfg 
Yep, you are right.

@Dhilan007 
Could you please fix it?

File: cocos/base/CCGeometry.cpp [hunk 1]
```diff
@@ -284,6 +284,15 @@ Point Point::getIntersectPoint(const Point& A, const Point& B, const Point& C, c
 }
 
 const Point Point::ZERO = Point(0, 0);
+const Point Point::ANCHOR_MIDDLE = Point(0.5, 0.5);
```
Comment (minggo) [inline cocos/base/CCGeometry.cpp:287]: 0.5 -> 0.5f.

**Comment by WenhaiLin** [comment]
@minggo 
yes,i should fix it

**Pull request merged**